.github/workflows/ci.yml

```
@@ -59,14 +61,29 @@ jobs:
               run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
 
             - name: Execute tests (Unit and Feature)
-              run: vendor/bin/phpunit
+              run: vendor/bin/phpunit --log-junit=reports/report-phpunit.xml --coverage-clover=reports/coverage-phpunit.xml
 
             - name: Static analysis with PHPStan
-              run: vendor/bin/phpstan analyse
+              run: |
+                vendor/bin/phpstan analyse --error-format=github
+                vendor/bin/phpstan analyse --error-format=json > reports/phpstan.json
 
             - name: Static analysis with Psalm
               if: "${{ matrix.experimental == false }}"
               run: vendor/bin/psalm
 
             - name: Coding style PSR12 Check
               run: vendor/bin/phpcs
+            
+            - name: 'Run SonarQube cloud scanner'
+              if: ${{contains(github.ref, '/pull/')}}
+              uses: SonarSource/sonarqube-scan-action@v5
+              env:
+                SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+            - name: 'Run SonarQube cloud scanner'
+              if: ${{!contains(github.ref, '/pull/')}}
+              uses: SonarSource/sonarqube-scan-action@v5
+              env:
+                SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+              with:
+                args: -Dsonar.branch.name=${{ github.ref_name }}
```

Goede suggestie van @jvanschie om gebruik te maken van de generic pipeline.

```suggestion
            - name: 'Run SonarQube cloud scanner'
              uses: minvws/nl-irealisatie-generic-pipelines/.github/actions/sonarcloud@main
              with:
                sonar-token: ${{ secrets.SONAR_TOKEN }}
```

---

Helaas gaat dat (nog) niet werken, want de [minvws/nl-irealisatie-generic-pipelines](https://github.com/minvws/nl-irealisatie-generic-pipelines) is een private repo en dit is een public repo. Dus of we gooien die andere eerst open of we doen het eerst met de uiteindelijke action zelf.

---

Als dat de way forward is, liever dat. Als het voor nu te veel werk is/geen tijd, dan graag een ticket richting Operations van de juiste persoon.

---

Dat kan volgens mij prima, moeten we alleen toestemming geven om die workflow uit de private-repo te gebruiken, maar aangezien ze beide in de minvws orga zitten moet dat kunnen.